src/Robot-Dobot_MG400/main.py

```diff
@@ -241,15 +255,29 @@ def apply(self):
         # print(x,y,z,r)
                     
         if self._last_xyzr != (x, y, z, r):
-            self._last_xyzr = (x, y, z, r)
+
+            if self.use_jump_mode:
+
+                # we only move to movement height if x, y, or r change
+                # in case of a z scan, we do not need to go back to movement height
+                if self._last_xyzr[0, 2] != (x, y) or self._last_xyzr[3] != r:
+
+                    # vertical move to movement height
+                    self.move_linear(self._last_xyzr[0], self._last_xyzr[1], self.movement_height, self._last_xyzr[3])
```

Comment:
Something that probably depends on the application, but what happens (or what should happen) when the movement_hight is already lower than the current position? Probably no need to change the code for now, but might be worth mentioning in the description.

Reply:
Basically both can work, but because of the geometry of the robot samples are typically approached from the top.
At the end the code is very generic. It will move from all positions to the movement height. This also makes sense. In case the robot starts (e.g. because of an accident) at the same height like the samples, it would go first back to movement height before approaching a new sample.
I will not add a description because each user needs to test this anyway for each individual setup.